qiskit_optimization/applications/bin_packing.py

```diff
@@ -0,0 +1,157 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An application class for the bin packing."""
+
+from typing import List, Union, Optional
+
+import numpy as np
+from docplex.mp.model import Model
+from qiskit.exceptions import MissingOptionalLibraryError
+
+from qiskit_optimization.algorithms import OptimizationResult
+from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+from qiskit_optimization.translators import from_docplex_mp
+from .optimization_application import OptimizationApplication
+
+try:
+    import matplotlib.pyplot as plt
+    from matplotlib.pyplot import Figure
+
+    _HAS_MATPLOTLIB = True
+except ImportError:
+    _HAS_MATPLOTLIB = False
+
+    class Figure:  # type: ignore
+        """Empty Figure class
+        Replacement if matplotlib.pyplot.Figure is not present.
```

Comment:
```suggestion
        Replacement Figure for when matplotlib is not present.
```
CI failed on pyplot - we could add pyplot to the custom dictionary - however matplotlib is already there, so maybe this rewording suggestion is fine.